go/vt/wrangler/reparent.go

```diff
@@ -1043,6 +1026,52 @@ func (wr *Wrangler) emergencyReparentShardLocked(ctx context.Context, ev *events
 	return nil
 }
 
+// CheckTabletIsFarthestAhead will take a tablet alias string, along with a statusMap of tablet alias strings to tablet Status objects,
+// and return an error if the tablet is not the farthest ahead, or otherwise return nil if it is the farthest ahead.
+func (wr *Wrangler) CheckTabletIsFarthestAhead(tabletAliasStr string, statusMap map[string]*replicationdatapb.Status) error {
```

Comment:
Why is this function exported? Do we expect to use it outside wrangler?

Reply:
Yes, we do. I think it's important that we make it publically available exactly how we are determining the farthest ahead tablet, since we require a user to supply us with a fully caught up tablet when issuing an emergency reparent. I've exported two functions - this one, and `FindMasterCandidates` which is a method that will find all of the tablets which are fully caught up. A user may decide which they may want to use, depending on how much control they need.

Reply:
Ah, operator use case. Makes sense.